gamemode/modules/fadmin/fadmin/access/sh_shared.lua

```diff
@@ -7,9 +7,16 @@ FAdmin.Access.ADMIN[0] = "user"
 FAdmin.Access.Groups = FAdmin.Access.Groups or {}
 FAdmin.Access.Privileges = FAdmin.Access.Privileges or {}
 
-function FAdmin.Access.AddGroup(name, admin_access --[[0 = not admin, 1 = admin, 2 = superadmin]], privs, immunity, fromCAMI)
+function FAdmin.Access.AddGroup(name, admin_access --[[0 = not admin, 1 = admin, 2 = superadmin]], privs, immunity, fromCAMI, CAMIsrc)
     FAdmin.Access.Groups[name] = FAdmin.Access.Groups[name] or {ADMIN = admin_access, PRIVS = privs or {}, immunity = immunity}
 
+    --Make sure things that come from CAMI come with a CAMIsrc
+    assert((fromCAMI and CAMIsrc ~= nil) or ((not fromCAMI) and CAMIsrc == nil))
+    --If the CAMIsrc is a string, save it, otherwise save an empty string
+    if type(CAMIsrc) ~= "string" then
```

Comment:
You can use the `isstring` function here.

Reply:
Done!